nni/compression/pytorch/utils/counter.py

```diff
@@ -26,110 +225,113 @@ def count_flops_params(model: nn.Module, input_size, custom_ops=None, verbose=Tr
     ---------
     model : nn.Module
         target model.
-    input_size: list, tuple
-        the input shape of data
+    x: tuple or tensor
+        the input shape of data or a tensor as input data
     custom_ops: dict
         a mapping of (module: custom operation)
         the custom operation will overwrite the default operation.
         for reference, please see ``custom_mask_ops``.
+    verbose: bool
+        If False, mute detail information about modules. Default is True.
+    mode:
+        the mode of how to collect information. If the mode is set to `default`,
+        only the information of convolution and linear will be collected.
+        If the mode is set to `full`, other operations will also be collected.
 
     Returns
     -------
     flops: float
         total flops of the model
     params:
         total params of the model
+    results: dict
+        detail information of modules
     """
 
-    assert input_size is not None
+    assert isinstance(x, tuple) or isinstance(x, torch.Tensor)
+    assert mode in ['default', 'full']
+
+    original_device = next(model.parameters()).device
+    training = model.training
 
-    device = next(model.parameters()).device
-    inputs = torch.randn(input_size).to(device)
+    if torch.is_tensor(x[0]):
```

Comment:
so here `x` is a list?

Reply:
yes, x are positional arguments given to the module